fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -1747,38 +1748,7 @@ public Void visitCreateFunctionStatement(CreateFunctionStmt statement, ConnectCo
 
         @Override
         public Void visitShowFunctionsStatement(ShowFunctionsStmt statement, ConnectContext context) {
-            // Don't do any privilege check on show global functions.
-            if (statement.getIsGlobal()) {
-                return null;
-            }
-            if (!PrivilegeActions.checkAnyActionOnDb(context, statement.getDbName())) {
-                ErrorReport.reportSemanticException(ErrorCode.ERR_DB_ACCESS_DENIED,
-                        context.getQualifiedUser(), statement.getDbName());
-            }
-            List<Table> tableList = GlobalStateMgr.getCurrentState().getDb(statement.getDbName()).getTables();
-            boolean hasPrivilege = false;
-            for (Table table : tableList) {
-                if (table.getType().equals(Table.TableType.VIEW)) {
-                    if (PrivilegeActions.checkAnyActionOnView(context, statement.getDbName(), table.getName())) {
-                        hasPrivilege = true;
-                        break;
-                    }
-                } else if (table.getType().equals(Table.TableType.MATERIALIZED_VIEW)) {
-                    if (PrivilegeActions.checkAnyActionOnMaterializedView(context, statement.getDbName(),
-                            table.getName())) {
-                        hasPrivilege = true;
-                        break;
-                    }
-                }
-                if (PrivilegeActions.checkAnyActionOnTable(context, statement.getDbName(), table.getName())) {
-                    hasPrivilege = true;
-                    break;
-                }
-            }
-            if (!hasPrivilege) {
-                ErrorReport.reportSemanticException(ErrorCode.ERR_ANY_ACTION_IN_DB_ACCESS_DENIED_ERROR,
-                        statement.getDbName());
-            }
+            // Don't do any privilege check on show functions
```

Comment:
I think this statement is inaccurate, actually we do, you can comment that we will do it in showExecutor in a more subtle way.

Reply:
ok, modify in next pr